instrumentor/controllers/instrumentationdevice/common.go

```diff
@@ -157,46 +148,46 @@ func removeInstrumentationDeviceFromWorkload(ctx context.Context, kubeClient cli
 		return errors.New("unknown kind")
 	}
 
-	err := kubeClient.Get(ctx, client.ObjectKey{
-		Namespace: namespace,
-		Name:      workloadName,
-	}, workloadObj)
-	if err != nil {
-		return client.IgnoreNotFound(err)
-	}
-
-	result, err := controllerutil.CreateOrPatch(ctx, kubeClient, workloadObj, func() error {
+	err := retry.RetryOnConflict(retry.DefaultRetry, func() error {
```

Comment:
I think we should verify that this behaves as we expect by doing a large batch of un-instrumentation and adding some logs here for debug